### PR TITLE
builtin: add lpad built-in function

### PR DESCRIPTION
**Pull request by hsqlu**
Add LPAD built-in function.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM
Good Job!

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1599,5 +1599,38 @@ type builtinLpadSig struct {
...
+	}
+	l := int(length)
+
+	padStr, err := args[2].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if l < 0 || (len(str) < l && padStr == "") {
+		d.SetNull()
```
Comment (zimulala) [inline expression/builtin_string.go]: Remove this line，because the default value of `d` is `null`.
Rest LGTM.

Comment (hsqlu) [inline expression/builtin_string.go]: Got that. I've removed that line. PTAL again, thanks!

**Comment by zimulala** [comment]
LGTM

**Pull request merged**